pkg/api/handlers/utils/images.go

```diff
@@ -76,8 +77,16 @@ func GetImages(w http.ResponseWriter, r *http.Request) ([]*image.Image, error) {
 		err    error
 	)
 
-	if len(query.Filters) > 0 {
-		for k, v := range query.Filters {
+	queryFilters := query.Filters
+	if len(query.Filter) > 0 { // Docker 1.24 compatibility
```

Comment:
Here, we need to make sure that we only accept this parameter for the docker-compat endpoints. There's a helper function to check if the request for libpod:
```golang
if !IsLibpodRequest(r) && en(query.Filter) > 0 { ...`

Reply:
Makes sense. Added this check.